hw/ip/flash_ctrl/doc/_index.md

```diff
@@ -65,40 +75,130 @@ The physical controller supports the following features
+*  Date partitions can be directly read by software and other hardware hosts, while information partitions can only be read by the flash controller
+
+By default, this design assumes 1 type of information partition and 4 pages per type of information partition.
 
 
 # Theory of Operation
 
 ## Block Diagram
 
+![Flash Block Diagram](flash_block_diagram.svg)
+
 ### Flash Protocol Controller
 
-The Flash Protocol Controller sits between the host software interface and the flash physical controller, which contains the physical flash.
-Its primary function is to translate software requests into a high level protocol for the actual flash block.
-Importantly, the flash protocol controller is not responsible for the detailed timing and waveform control of the flash.
-Instead, it maintains FIFOs / interrupts for the software to process data, as well as high level abstraction of region protection controls.
+The Flash Protocol Controller sits between the host software interface, other hardware components and the flash physical controller.
+Its primary functions are two fold
+*  Translate software program, erase and read requests into a high level protocol for the actual flash physical controller
+*  Act as communication interface between flash and other components in the system, such as life cycle and key manager.
+
+Importantly, the flash protocol controller is not responsible for the detailed timing and waveform control of the flash, nor is it responsible for data scrambling and reliability metadata such as parity and ECC.
+Instead, it maintains FIFOs / interrupts for the software to process data, as well as high level abstraction of region protection controls as well as error handling.
+
+There are two hardware connections between the flash controller and the rest of the design.
+*  The first is to life cycle, in which flash content plays a part.
+*  The second is to key manager, whose seed material may be partially sourced from flash.
 
 ### Flash Physical Controller
 
-The Flash Physical Controller is the wrapper module that contains the actual vendor flash memory instantiation.
-It is responsible for arbitrating high level protocol commands (such as read, program, erase) as well as any additional security and reliability features.
+The Flash Physical Controller is the wrapper module that contains the actual flash memory instantiation.
+It is responsible for arbitrating high level protocol commands (such as read, program, erase) as well as any additional security (scrambling) and reliability (ECC) features.
 The contained vendor wrapper module is then responsible for converting high level commands into low level signaling and timing specific to a particular flash vendor.
 The vendor wrapper module is also responsible for any BIST, redundancy handling, remapping features or custom configurations required for the flash.
 
-The diagram below summarizes the high level breakdown.
+The flash physcial controller supports the flash scramble function.
```

Comment:
typo: physcial
Maybe discard this line (the second sentence already mentions this)?

Reply:
fixed